### PR TITLE
Turn off remote cache for local dev, cleanup github actions UI clutter

### PR DESCRIPTION
**Pull request by jerrymarino**
Configure bazelrc in the `preflight_env.sh` script

Ensure we enable existing remote cache on github actions

### CODE REVIEW

**Comment by thiagohmcruz** [comment]
Add @luispadron and @mattrobmattrob since they know this workflow better than me

File: .bazelrc [hunk 1]
```diff
@@ -54,19 +54,17 @@ build:ios_multi_arch_test --ios_minimum_os=10.2
...
-build:remote_cache --experimental_remote_cache_compression
-build:remote_cache --experimental_remote_merkle_tree_cache
-build:remote_cache --modify_execution_info=^(BitcodeSymbolsCopy|BundleApp|BundleTreeApp|DsymDwarf|DsymLipo|GenerateAppleSymbolsFile|ObjcBinarySymbolStrip|CppLink|ObjcLink|ProcessAndSign|SignBinary|SwiftArchive|SwiftStdlibCopy)$=+no-remote,^(BundleResources|ImportedDynamicFrameworkProcessor)$=+no-remote-exec
-build:remote_cache --nolegacy_important_outputs
-build:remote_cache --noslim_profile
-build:remote_cache --remote_cache=grpcs://remote.buildbuddy.io
-build:remote_cache --remote_timeout=3600
-
-# Enable remote_cache config by default
-build --config=remote_cache
```
[the remark this answers is not in the document]
Comment (jerrymarino) [reply .bazelrc:69]: Yep that's a good way to put it - the new streamlined stuff works on github only.

[the remark this answers is not in the document]
Comment (jerrymarino) [reply .bazelrc:69]: @mattrobmattrob inside of `/path/to/rules_ios` you can add that into `user.bazelrc` if you'd like, rules_ios conditionally imports it.

**Comment by jszumski** [comment]
What is the functional change implemented here? Just defaulting the cache off for local dev?

File: .bazelrc (hunk 1 above)
[the remark this answers is not in the document]
Comment (luispadron) [reply .bazelrc:69]: I guess im confused why the default is to not enable the cache?

If the issue is out internal forks shouldn't hit buildbuddy then why don't we disable it there?

[the remark this answers is not in the document]
Comment (jerrymarino) [reply .bazelrc:69]: I don't think it is a good default from security standpoint - and I don't maintain forks of rules_ios to disable something like this

[the remark this answers is not in the document]
Comment (jerrymarino) [reply .bazelrc:69]: Or any forks of rules_ios BTW 😉

File: .github/workflows/preflight_env.sh [hunk 2]
```diff
@@ -6,3 +6,7 @@ printenv
 
 sudo xcode-select -p
 sudo xcode-select -s /Applications/Xcode_14.2.app
+
+cat << EOF > $HOME/.bazelrc
```
Comment (mattrobmattrob) [inline .github/workflows/preflight_env.sh:10]: Seems it would make sense to use `$HOME/user.bazelrc` in this case.

**Review by mattrobmattrob — APPROVED** [review]

**Comment by jerrymarino** [comment]
For now I've mitigated the hole for us when running locally. Dropping the `-s` in github output because it breaks looking at CI output in the UI me so also dropped it.

**Comment by jerrymarino** [comment]
For now I've fixed it when running locally. Also - dropping the `-s` because it breaks looking at CI output in the UI me

**Pull request merged**